### PR TITLE
Compare purchase urgency

### PR DESCRIPTION
**Pull request by Terieyenike**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description

Sorting items by how soon the user might need to buy them will make it easier for the user to identify what they need to buy, which will in turn help them better plan their shopping trips. To that end, we will be sorting items into 4 possible groups, according to how urgently the user should buy the item:

1. Need to buy soon (7 days or fewer until the next purchase)
2. Need to buy kind of soon (between 7 & 30 days until the next purchase)
3. Need to buy not soon (30 days or more until the next purchase)
4. Inactive (60 days have passed since the last purchase)

To implement this feature, your team will need to create some new UI elements so the user knows how their items are sorted and write some logic behind-the-scenes to perform the sorting itself. Read the acceptance criteria and our notes carefully to be sure you understand the tasks at hand, and don’t be afraid to reach out to the rest of your team, and your mentors if you need help. You’ve got this! 

## Acceptance Criteria

- [x] Items in the list are shown with an indicator that tells the user they should buy the item “soon”, “kind of soon”, or “not soon”; or that the item is “inactive”
	- [x] This urgency indicator *does not* rely on only color
- [ ] `api/firestore.js` exports a new `comparePurchaseUrgency` function with the following behaviors
	- [ ] sorts inactive items last, then
	- [ ] sorts items in ascending order of days until purchase, and
	- [ ] sorts items with the same days until purchase alphabetically

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### CODE REVIEW

**Comment by Terieyenike** [comment]
@thatgirldorian help us check this PR.

Thank you!

**Comment by pmbanugo** [comment]
What issue does is this related to?

**Comment by thatgirldorian** [comment]
> What issue does is this related to?

Hi Peter! This PR is related to issue 12, but we've merged our code on the other PR and will remove this after that one has been removed.

File: src/views/List.jsx [hunk 1]
```diff
@@ -14,6 +15,8 @@ export function List({ data, listToken }) {
 		name.toLowerCase().includes(searchField),
 	);
 
+	const sortedList = comparePurchaseUrgency(data);
+
```
Comment (faysvas) [inline src/views/List.jsx:19]: Instructions for testing this PR:
`comparePurchaseUrgency` accepts data that represents an unsorted list with the items that a user has added, and returns a list sorted according to the acceptance criteria of issue 12. 
To test this create a list (or join an existing one) with items that have a variety of `dateNextPurchased` dates and one or more inactive items (`dateLastPurchased` is over 60 days ago). A sample list you can join is the fifth one in firestore which starts with "co...".
So the way to compare the lists is adding `console.log(data)` above line 18,  and then add `console.log(sortedList)` under line 18.

**Review by pmbanugo — APPROVED** [review]

**Pull request closed without merging**